src/read_pool.rs

```diff
@@ -154,14 +157,18 @@ pub struct ReadPoolRunner<E: Engine, R: FlowStatsReporter> {
     engine: Option<E>,
     reporter: R,
     inner: FutureRunner,
+    after_start: Option<Arc<Callback>>,
```

Comment:
Maybe we don't need a generic `after_start` here. I think it's ok to just put a registry here and register in the start method.